create.go

```diff
@@ -1,14 +1,121 @@
-package statusresouce
+package statusresource
 
 import (
 	"context"
+	"encoding/json"
+	"time"
+
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	clusterStatus, err := r.clusterStatusFunc(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var patches []Patch
+
+	// In case a CR might not have a status we can work with we have to initialize
+	// it.
+	//
+	// TODO remove this as soon as all CRs have a status structure applied.
```

Comment:
I would just keep this, it should be fine to add back the status structure if it was removed. Kind of like reconciling the CR structure.

Reply:
Yeah I was thinking about this. It is only that the condition to check if it is empty is a bit odd. For that all of our structures should actually be zero valued to `nil` which means we either define pointers or slices. That is bigger though and I just went with this now. Lets see.